app/controllers/sessions_controller.rb

```diff
@@ -3,16 +3,29 @@ class SessionsController < ApplicationController
 
   def create
     auth_hash = request.env['omniauth.auth']
+
     if auth_hash["uid"]
       @user = User.find_or_create_user(auth_hash)
+
+      if auth_hash[:credentials][:token] != nil
+        session[:access_token] = auth_hash[:credentials][:token]
```

Comment:
This is just for instagram, yeah? It'd be worth commenting when you would and wouldn't expect this token to exist.